src/gui/guiFormSpecMenu.cpp

```diff
@@ -4414,7 +4369,6 @@ bool GUIFormSpecMenu::OnEvent(const SEvent& event)
 		// If m_selected_amount has been decreased to zero, deselect
 		if (m_selected_amount == 0) {
 			m_selected_swap.clear();
-			delete m_selected_item;
 			m_selected_item = nullptr;
```

Comment:
Because this is not a plain pointer please use `m_selected_item.reset();` here.

Reply:
Done.